clang/test/Index/Core/index-source.cpp

```diff
@@ -285,30 +285,23 @@ template<>
 class SpecializationDecl<int>;
 // CHECK: [[@LINE-1]]:7 | class(Gen,TS)/C++ | SpecializationDecl | c:@S@SpecializationDecl>#I | <no-cgname> | Decl,RelSpecialization | rel: 1
 // CHECK-NEXT: RelSpecialization | SpecializationDecl | c:@ST>1#T@SpecializationDecl
-// CHECK: [[@LINE-3]]:7 | class(Gen,TS)/C++ | SpecializationDecl | c:@S@SpecializationDecl>#I | <no-cgname> | Ref | rel: 0
```

Comment:
Can you explain what is going on here that these lines are being lost?

Reply:
The "type as written" for a specialization is no longer being indexed as a reference to that specialization. The `TemplateArgumentLoc`s still are, but I need to add the corresponding `CHECK` lines so we know for sure

Reply:
@erichkeane updated the tests to check for references within `TemplateArgumentLoc`s